packages/app-desktop/gui/ConfigScreen/FontSearch.tsx

```diff
@@ -0,0 +1,82 @@
+import React = require('react');
+import { useMemo, useState, useCallback } from 'react';
+import { CSSProperties } from 'styled-components';
+
+interface Props {
+	_key: string;
+	updateSettingValue: (key: string, value: string)=> void;
+	inputType: string;
+	inputStyle: CSSProperties;
+	fieldValue: string;
+	fonts: string[];
+}
+
+const FontSearch = (props: Props) => {
+	const { _key: key, updateSettingValue, inputType, inputStyle, fieldValue, fonts } = props;
+	const [inputText, setInputText] = useState('');
+	const areFontsLoading = fonts.length === 0;
+
+	const filteredFonts = useMemo(() => {
+		return fonts.filter((font: string) =>
+			font.toLowerCase().startsWith(inputText.toLowerCase()),
+		);
+	}, [fonts, inputText]);
+
+	const showList = useCallback((display: boolean) => {
+		document.getElementById(`search-list-${key}`).style.display = display ? 'block' : 'none';
+	}, [key]);
+
+	const onTextChange = useCallback((event: any) => {
+		setInputText(event.target.value);
+		updateSettingValue(key, event.target.value);
+		showList(true);
+	}, [key, updateSettingValue, showList]);
+
+	const onFocusHandle = useCallback((event: any) => {
+		setInputText(event.target.value); // To handle the case when the value is already set
+		showList(true);
+	}, [showList]);
+
+	const onBlurHandle = useCallback(() =>
+		setTimeout(() => showList(false), 150) // Delay the hiding of the list to allow the click event to fire
+	, [showList]);
+
+	const onFontClick = useCallback((event: any) => {
+		const font = event.target.innerText;
+		(document.getElementById(key) as HTMLDivElement).innerText = font;
+		updateSettingValue(key, font);
+		showList(false);
+	}, [key, updateSettingValue, showList]);
+
+	return (
+		<>
+			<input
+				type={inputType}
+				style={inputStyle}
+				value={fieldValue}
+				id={key}
```

Comment:
Please remove the ID. Setting and accessing global IDs is not the way to work in React as they can conflict with other IDs.